### PR TITLE
CSHARP-5420: Driver Support for Search Sequential Pagination

### PR DESCRIPTION
**Pull request by papafe**

### CODE REVIEW

**Review by adelinowona — CHANGES_REQUESTED** [review]

File: src/MongoDB.Driver/ProjectionDefinitionBuilder.cs [hunk 1]
```diff
@@ -257,6 +257,40 @@ public static ProjectionDefinition<TDocument> MetaSearchScoreDetails<TDocument>(
...
+        /// <param name="field">The field.</param>
+        /// <returns>
+        /// A combined projection.
+        /// </returns>
+        public static ProjectionDefinition<TDocument> MetaSearchSequenceToken<TDocument>(
+            this ProjectionDefinition<TDocument> projection,
+            FieldDefinition<TDocument> field)
+        {
+            var builder = Builders<TDocument>.Projection;
+            return builder.Combine(projection, builder.MetaSearchScoreDetails(field));
```
[the remark this answers is not in the document]
Comment (papafe) [reply src/MongoDB.Driver/ProjectionDefinitionBuilder.cs:274]: Yes 🤦

File: src/MongoDB.Driver/IAggregateFluent.cs [hunk 2]
```diff
@@ -404,6 +404,7 @@ IAggregateFluent<TNewResult> Lookup<TForeignDocument, TAsElement, TAs, TNewResul
         IAggregateFluent<BsonDocument> SetWindowFields<TWindowFields>(
             AggregateExpressionDefinition<ISetWindowFieldsPartition<TResult>, TWindowFields> output);
 
+        //TODO If I add a parameter here, then this would be a binary breaking change
```
[the remark this answers is not in the document]
Comment (papafe) [reply src/MongoDB.Driver/IAggregateFluent.cs:407]: I completely agree that users should use `SearchOptions`, but given we have the other parameters here I thought we would want to add it here. 
I got no issues about not adding parameters here :)

[the remark this answers is not in the document]
Comment (adelinowona) [reply src/MongoDB.Driver/IAggregateFluent.cs:407]: @BorisDog what do you think?

File: src/MongoDB.Driver/ProjectionDefinitionBuilder.cs [hunk 3]
```diff
@@ -576,7 +610,7 @@ public ProjectionDefinition<TSource> Include(Expression<Func<TSource, object>> f
         /// <param name="field">The field.</param>
         /// <param name="metaFieldName">The meta field name.</param>
         /// <returns>
-        /// A text score projection.
+        /// A meta projection.
```
Comment (BorisDog) [inline src/MongoDB.Driver/ProjectionDefinitionBuilder.cs:613]: 👍

File: src/MongoDB.Driver/IAggregateFluent.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (BorisDog) [reply src/MongoDB.Driver/IAggregateFluent.cs:407]: We could add this overload `IAggregateFluentExtensions` if needed, but I don't see much value in that.
In first Search implementation we've started with only a few params, but not it becomes untastable to keep adding them directly. So I'd prefer to stick to `SearchOptions`.

**Review by adelinowona — APPROVED** [review]
LGTM

**Pull request merged**